docs(diagnostics): add quickfix handler example (#31371)
neovim/neovim: runtime/doc/diagnostic.txt

    }
<

                                          *diagnostic-loclist-example*
Whenever the |location-list| is opened, the following `show` handler will show
the most recent diagnostics: >lua

  vim.diagnostic.handlers.loclist = {
    show = function(_, _, _, opts)
      -- Generally don't want it to open on every update
      opts.loclist.open = opts.loclist.open or false
      vim.diagnostic.setloclist(opts.loclist)
    end
  }
<

> The handler accepts the same options as |vim.diagnostic.setloclist()| and can be
> configured using |vim.diagnostic.config()|: >lua
> 
>   vim.diagnostic.config({
>     loclist = {
>       open = true,
>       severity = { min = vim.diagnostic.severity.WARN },
>     }
>   })
> <
> 
> This will open the location list window on every diagnostic change, and only
> list warnings and errors.

Reviewer: ```suggestion
The handler accepts the same options as |vim.diagnostic.setloclist()| and can be
configured using |vim.diagnostic.config()|: >lua

  -- Open the location list on every diagnostic change (warnings/errors only).
  vim.diagnostic.config({
    loclist = {
      open = true,
      severity = { min = vim.diagnostic.severity.WARN },
    }
  })
<
```

    }
<

>                                           *diagnostic-handlers-example-loclist*
> To automatically update the |location-list| list with new diagnostics, we can
> create a new handler: >lua
> 
>   vim.diagnostic.handlers.loclist = {
>     show = function(_, _, _, opts)
>       -- Generally don't want it to open on every update
>       opts.loclist.open = opts.loclist.open or false
>       vim.diagnostic.setloclist(opts.loclist)
>     end
>   }
> <
> 
> The "hide" function can be omitted for this handler.
> 
> Now, everytime the |location-list| is opened, it will show the most recent
> diagnostics. The handler accepts the same options as
> |vim.diagnostic.setloclist()| and can be configured using
> |vim.diagnostic.config()|: >lua

Reviewer: This is thoughtfully worded, but still way too verbose. Examples should be as small as possible, and only essential things should be noted. Otherwise it just gets messy and hard to read and maintain.

```suggestion
                                          *diagnostic-loclist-example*
Whenever the |location-list| is opened, the following `show` handler will show
the most recent diagnostics: >lua

  vim.diagnostic.handlers.loclist = {
    show = function(_, _, _, opts)
      -- Generally don't want it to open on every update
      opts.loclist.open = opts.loclist.open or false
      vim.diagnostic.setloclist(opts.loclist)
    end
  }
<

The handler accepts the same options as |vim.diagnostic.setloclist()| and can be
configured using |vim.diagnostic.config()|: >lua
```
